ducktape/tests/session_context.py

```diff
@@ -83,4 +134,75 @@ def generate_results_dir(session_id):
     :type session_id: str
     :rtype: str
     """
-    return session_id + "-test-results"
\ No newline at end of file
+    return session_id + "-test-results"
+
+
+def mkdir_p(path):
```

Comment:
This function is duplicated. It should probably be refactored into some utils module. That module should probably be clearly named to indicate it's for local utilities since the stuff in RemoteAccount and utilities like this are going to be easy to mix up.